src/loaders/OmeZarrLoader.ts

```diff
@@ -105,84 +105,47 @@ class SmartStoreWrapper implements AsyncStore<ArrayBuffer, RequestInit> {
   requestQueue?: RequestQueue;
-  // Our other miscellaneous info is packed into an object to be null-checked all at once
-  arrayInfo?: {
-    axesTCZYX: [number, number, number, number, number];
-    scaleNames: string[];
-    dimensionSeparator: string;
-  };
-
-  constructor(baseStore: AsyncStore<ArrayBuffer, RequestInit>, requestQueue?: RequestQueue) {
+
+  constructor(baseStore: AsyncStore<ArrayBuffer, RequestInit>, cache?: VolumeCache, requestQueue?: RequestQueue) {
     this.baseStore = baseStore;
+    this.cache = cache;
     this.requestQueue = requestQueue;
     this.listDir = baseStore.listDir;
     this.rmDir = baseStore.rmDir;
     this.getSize = baseStore.getSize;
     this.rename = baseStore.rename;
   }
 
-  setCacheInfo(
-    axesTCZYX: [number, number, number, number, number],
-    scaleNames: string[],
-    dimensionSeparator = "/",
-    cache?: VolumeCache,
-    cacheStore?: CacheStore
-  ) {
-    this.arrayInfo = { axesTCZYX, scaleNames, dimensionSeparator };
-    this.cache = cache;
-    this.cacheStore = cacheStore;
-  }
-
-  private async cacheWhenReceived(prom: Promise<ArrayBuffer>, dims: Partial<DataArrayExtent>): Promise<ArrayBuffer> {
-    const result = await prom;
-    if (this.cache && this.cacheStore) {
-      this.cache.insert(this.cacheStore, result, dims);
+  private async getAndCacheItem(item: string, cacheKey: string, opts?: RequestInit): Promise<ArrayBuffer> {
+    const result = await this.baseStore.getItem(item, opts);
+    if (this.cache) {
+      this.cache.insert(cacheKey, result);
     }
     return result;
   }
 
-  async getItem(item: string, opts?: RequestInit | undefined): Promise<ArrayBuffer> {
+  getItem(item: string, opts?: RequestInit): Promise<ArrayBuffer> {
     // If we don't have a cache or aren't getting a chunk, call straight to the base store
     const zarrExts = [".zarray", ".zgroup", ".zattrs"];
-    if (!this.cache || !this.cacheStore || !this.arrayInfo || zarrExts.some((s) => item.endsWith(s))) {
+    if (!this.cache || zarrExts.some((s) => item.endsWith(s))) {
       return this.baseStore.getItem(item, opts);
     }
 
-    const { scaleNames, axesTCZYX, dimensionSeparator } = this.arrayInfo;
-
-    // Extract chunk coordinates from item path
-    const pathElems = item.split(dimensionSeparator);
-    // The first element should be the scale name - convert it to scale level index
-    const scale = scaleNames.indexOf(pathElems.shift() as string);
-    // The remaining elements should be chunk coordinates, and should be numbers
-    const coordsLength = getDimensionCount(axesTCZYX);
-    const chunkCoords = pathElems.slice(-coordsLength).map((s) => parseInt(s, 10));
-    if (chunkCoords.length < coordsLength || chunkCoords.some(isNaN) || scale < 0) {
-      console.log("zarr store wrapper: unexpected item path: " + item);
-      return this.baseStore.getItem(item, opts);
-    }
+    const keyPrefix = (this.baseStore as HTTPStore).url || "";
+    const key = keyPrefix + item;
```

Comment:
Nit: I wonder if a separator character might make it more human readable, like `?`, `/` or `::`